app/controllers/admin/base_controller.rb

```diff
@@ -93,9 +93,16 @@ def batch
     raise 'No ids specified for batch action' unless params[:by_ids].present?
     batch_action = params[:batch_action].to_sym
     if allow_batch_action?(batch_action) && collection.all?{|item| can?(batch_action, item) }
-      count = collection.inject(0) { |c, item| apply_batch_action(item, batch_action) ? c + 1 : c }
-      batch_action_name = I18n.t("admin.actions.batch_#{batch_action}.title", default: batch_action.to_s.humanize)
-      flash[:success] = I18n.t('admin.batch_actions.status', count: count, action: batch_action_name)
+      batch_action_collection_method = :"#{batch_action}_collection"
+      if resource_class.respond_to?(batch_action_collection_method)
```

Comment:
It should be better to check method name instead of reflection.
For example: `batch_action.to_s.end_with?('_collection')`.